crates/rspack/tests/tree-shaking/reexport-all-as/expected/main.js

```diff
@@ -6,6 +6,7 @@ __webpack_require__.d(__webpack_exports__, {
   'a': function() { return a; }
 });
  function a() {}
+ function dddd() {}
```

Comment:
We added a extra variable, and only used declaration `a` , https://github.com/web-infra-dev/rspack/pull/3800/files#diff-88e9e2f5deab4237bd10fad2b7a74adc310be0e3b4e1a902f29c59e5f0f40687R34, so the `dddd` is not exported as expected.